plugins/services/src/js/components/modals/CreateServiceModalForm.js

```diff
@@ -309,7 +319,18 @@ class CreateServiceModalForm extends Component {
       baseConfigCopy
     );
 
-    return CreateServiceModalFormUtil.stripEmptyProperties(newConfig);
+    const { labels, ...config } = newConfig;
+
+    if (Object.keys(labels).length === 0) {
+      return {
+        ...CreateServiceModalFormUtil.stripEmptyProperties(config)
+      };
+    }
+
+    return {
+      labels,
```

Comment:
why are we destructing it above, only to merge it together again? Is this a way to fix a broken `CreateServiceModalFormUtil.stripEmptyProperties`?

Reply:
`return CreateServiceModalFormUtil.stripEmptyProperties(newConfig)`?!